src/Components/Users/UserProfile.tsx

```diff
@@ -186,9 +187,33 @@ export default function UserProfile() {
         }
     };
 
-    return (
+    const checkForNewBuildVersion = async () => {
+
+        let [isLatestBundle, newVersion] = await checkIfLatestBundle();
+
+        if (!isLatestBundle) {
+            setUpdateBtnText("updating...");
+            localStorage.setItem('build_meta_version', newVersion);
+
+            if ('caches' in window) {
+                // Service worker cache should be cleared with caches.delete()
+                caches.keys().then(names => {
+                    for (const name of names) {
+                        caches.delete(name);
+                    }
+
+                    window.location.reload(true);
+                });
+            }
+        } else {
+            setUpdateBtnText("You already have the latest version!")
 
-        <div >
+            setTimeout(() => setUpdateBtnText("Update"), 1000);
```

Comment:
Codacy found an issue: [The function setTimeout can be unsafe](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5837957)